src/utils/paramsMapper.ts

```diff
@@ -0,0 +1,58 @@
+import { IDataExchange, IQueryParams } from './types/dataExchange';
+
+/**
+ * Map the value from an exchange trigger to a representation
+ */
+export const paramsMapper = async (props: {
+    resource: string;
+    representationQueryParams: string[];
+    dataExchange: IDataExchange;
+    url: string;
+}) => {
+    const { representationQueryParams, dataExchange, resource } = props;
+    let { url } = props;
+
+    //if providerParams exists use it
+    if (
+        dataExchange?.providerParams?.query &&
+        dataExchange?.providerParams?.query?.length > 0
+    ) {
+        url = `${url}${filterAndStringify({
+            query: dataExchange?.providerParams?.query,
+            representationQueryParams,
+        })}`;
+    }
+    //else find resource params
+    else {
+        const resourceParams = dataExchange?.resources.find(
+            (element) => element?.resource === resource
+        );
+        if (!resourceParams || !resourceParams.params) {
+            return { url };
+        }
+
+        url = `${url}${filterAndStringify({
+            query: resourceParams?.params?.query,
+            representationQueryParams,
+        })}`;
+    }
+    return { url };
+};
+
+const filterAndStringify = (props: {
+    query: [IQueryParams];
+    representationQueryParams: string[];
+}) => {
+    const filteredArray = props?.query?.filter((obj) => {
+        const key = Object.keys(obj)[0];
+        return props?.representationQueryParams?.includes(key);
+    });
+
+    return filteredArray
+        ?.map((obj) => {
+            const key = Object.keys(obj)[0];
+            const value = obj[key];
+            return `?${encodeURIComponent(key)}=${encodeURIComponent(value)}`;
+        })
+        .join('&');
```

Comment:
Hi !

Very interesting PR, with new perspectives on different use cases. 
I notified a small issue on the construction of the URL with the query parameters. 
It seems like the actual code is working only for a single query parameter; the url is constructed the following way : 

`<url_endpoint>?<key1>=<value1>&?<key2>=<value2>&?<key3>=<value3>`
instead of : 
`<url_endpoint>?<key1>=<value1>&<key2>=<value2>&<key3>=<value3>`.

Here is a suggested edit to fix this issue : 

```suggestion
    return '?' + filteredArray
        ?.map((obj) => {
            const key = Object.keys(obj)[0];
            const value = obj[key];
            return `${encodeURIComponent(key)}=${encodeURIComponent(value)}`;
        })
        .join('&');
```

Feel free to contact me if you need further information on the raised issue.

Many thanks for this great work, very useful for the PTX use cases!

Guillaume